
video_num: 0

################################################################################################## 
#                                     origin & Waypoints                                         #
#                            data type: [latitude, longitude, altitude]                          #
##################################################################################################

origin: [35.06965949, 128.57880311, 50.921] # 창원

waypoints: [
  [35.069379042854045, 128.57892811844124, 50.921], # 전역 0
  [35.06951081029883, 128.57894371404203, 50.921],  # 전역 1
  [35.069627918431856, 128.57889587909511, 50.921], # 전역 2
  [35.069379042854045, 128.57892811844124, 50.921], # 전역 0
]

goal_range: 2 # [m] # float이면 터짐 수정 필요

################################################################################################## 
#                                          Docking                                               #
################################################################################################## 

target_shape: 'Circle'
color_to_detect: 'red'
stop_area_threshold: 20000

################################################################################################## 
#                                          Autonomous                                            #
##################################################################################################

angle_num: 20
yaw_range: 30
range: 3

decline_factor: 0.6
min_scaling: 0.4

margin: 0.25  # 장애물 마진
safe_area:    # 선박 안전 영역(사각형)
  w: 1.0
  h: 1.4

################################################################################################## 
#                                             PredockCal                                         #
################################################################################################## 

front_angle_range: [-5, 5]
right_angle_range: [-120, -60]
left_angle_range: [60, 120]


# front
front_distance_1: 2.9
r_p_11: 2.0
r_p_12: 3.0
turn_time: 2.5

# right
front_distance_2: 1.4
r_p_21: 1.2
r_p_22: 1.7
wait_time: 3.0

# left
front_distance_3: 5.5  # 1 = 6, 2 = 5.5, 3 = 5 
r_p_31: 1.2
r_p_32: 1.7


##################################################################################################
#                                      Steering Controller                                       #
##################################################################################################

servo_range: [80, 110] # middle: 95
delta_dot_max: 2.1 # [rad/s] 2.09

##################################################################################################
#                                           Thruster                                             #
##################################################################################################

# 1600
thruster: 1600
kp_servo : 1.7
kd_servo : 0.3

# 1650
# thruster: 1650
# kp_servo : 1.3
# kd_servo : 0.3

# 1650
# thruster: 1700
# kp_servo : 1.3
# kd_servo : 0.3

##################################################################################################
#                                            STController                                        #
##################################################################################################
# Stop
servo_p_stop: 95
servo_s_stop: 95
thruster_p_stop: 1500
thruster_s_stop: 1500

# Turn Left
servo_p_turn_left: 120
servo_s_turn_left: 120
thruster_p_turn_left: 1570 
thruster_s_turn_left: 1570

# Drift Left
angle_adjustment_max: 15
distance_adjustment_max: 70
thruster_p_drift_left: 1570 
thruster_s_drift_left: 1570 

# Turn Right
servo_p_turn_right: 124
servo_s_turn_right: 70
thruster_p_turn_right: 1440 
thruster_s_turn_right: 1560 

# Drift Right
thruster_p_drift_right: 1570 
thruster_s_drift_right: 1570

# Go
servo_p_go: 95
servo_s_go: 95
thruster_p_go: 1570 
thruster_s_go: 1570 





